### ISSUE TITLE
Incorrect bounding rects returned from pygame.draw.rect

### ISSUE THREAD

**Issue by charlesej** [pygame/pygame#1305]
The bounding rects returned from `pygame.draw.rect` are incorrect in some instances. The following code gives an example of an incorrect bounding rect.

```python
import pygame


def create_bounding_rect(surface, start, surf_color):
    """Create a rect to bound all the pixels that don't match surf_color."""
    width, height = surface.get_clip().size
    xmin, ymin = width, height
    xmax, ymax = -1, -1

    for y in range(height):
        for x in range(width):
            if surface.get_at((x, y)) != surf_color:
                xmin = min(x, xmin)
                xmax = max(x, xmax)
                ymin = min(y, ymin)
                ymax = max(y, ymax)

    if -1 == xmax:
        # No points means 0 sized rect with the position at the start.
        return pygame.Rect(start, (0, 0))

    return pygame.Rect((xmin, ymin), (xmax - xmin + 1, ymax - ymin + 1))


surf_color = pygame.Color('black')
surface = pygame.display.set_mode((300, 300))
surface.fill(surf_color)
rect = pygame.Rect((0, 0), (50, 50))
rect.center = surface.get_rect().center

bounding_rect = pygame.draw.rect(surface, pygame.Color('red'), rect, 10)
actual_rect = create_bounding_rect(surface, rect.topleft, surf_color)

print('{} reported bounding rect'.format(bounding_rect))
print('{} actual bounding rect'.format(actual_rect))
```

The above code gives the following output showing the actual bounding rect is different than the bounding rect returned from `pygame.draw.rect`. The rect being drawn is bigger than what is being reported as drawn.

```
pygame 2.0.0.dev3 (SDL 2.0.10, python 3.7.4)
Hello from the pygame community. https://www.pygame.org/contribute.html
<rect(125, 125, 50, 50)> reported bounding rect
<rect(121, 121, 59, 59)> actual bounding rect
```

System details:
- os: windows 10 (64bit)
- python: 3.7.4 (64bit)
- pygame: 2.0.0.dev3 (SDL: 2.0.10) at fb621be23daecdd8fde85918896568c723d201bf

**Comment by charlesej** [comment]
Test added with `@unittest.expectedFailure` decorator via #1309.

**Comment by charlesej** [comment]
Resolved by #1494.
